pkg/operator/deploy/staticresources/master/deployment.yaml.tmpl

```diff
@@ -36,6 +36,13 @@ spec:
         ports:
         - containerPort: 8080
           name: http
+        resources:
+          limits:
```

Comment:
The master operator deployment runs a lot of controllers (and it's expected that customer workloads are not typically running on master nodes), so I am of the opinion that we probably shouldn't impose any limits on the master deployment. Do we know how/if other OpenShift operators do requests and limits?

Reply:
It should probably still have limits even if it is running on the master. just in case of a memory leak it avoids crashing the whole master node. having it that low might not be super optimal though.

Reply:
I checked over the shared cluster today (specifically at openshift-machine-api operators/pods that run on masters) and see that there are no limits set. There are requests, though, which I agree we should have.

This is just my 0.02 though from past experience. If we need limits, we should just set them a lot higher.

Reply:
I'm +1 with Caden here. Requests are generally considered very healthy, and allow the underlying OS to appropriately trigger things like eviction when resources are high on a node. Limits cause strange behavior, like restarting containers and crashlooping when memory is high, and unnecessary throttling on CPU resources (which can lead to outages, like if your context will timeout 100% of the time if you don't get enough cycles).

The only real benefit we get to setting Limits is getting put in the QoS tier of Guaranteed, which I don't think is helpful when other (honestly, more critical) operators are NOT in this class as Caden has discovered (meaning, we'd end up evicting other, more critical components before our operator if we achieve Guaranteed...): https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/#create-a-pod-that-gets-assigned-a-qos-class-of-guaranteed

Reply:
requests have nothing to do with the os, it is for the scheduler